custom/abdm/auth.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+
+from custom.abdm.models import ABDMUser
+
+
+class ABDMUserAuthentication(TokenAuthentication):
+
+    def authenticate_credentials(self, token):
+        if not token:
+            return None, None
+        try:
+            user = ABDMUser.objects.get(access_token=token)
+        except ABDMUser.DoesNotExist:
+            raise AuthenticationFailed('Unauthorized')
+        if user.is_token_expired:
+            raise AuthenticationFailed('Token expired. Please re-sync.')
```

Comment:
These two exceptions should return different HTTP codes ideally. But if the callers don't care about it, that's fine.